python/semantic_kernel/connectors/ai/mistral_ai/services/mistral_ai_text_embedding.py

```diff
@@ -0,0 +1,112 @@
+
+    client: MistralAsyncClient
+
+    def __init__(
+        self,
+        ai_model_id: str | None = None,
+        api_key: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        client: MistralAsyncClient | None = None,
+    ) -> None:
+        """Initialize the Mistral AI Text Embedding service.
+
+        If no arguments are provided, the service will attempt to load the settings from the environment.
+        The following environment variables are used:
+        - MISTRALAI_API_KEY
+        - MISTRALAI_EMBEDDING_MODEL_ID
+
+        Args:
+            ai_model_id: (str | None): A string that is used to identify the model such as the model name. 
+            api_key (str | None): The API key for the Mistral AI service deployment. 
+            service_id (str | None): Service ID for the embedding completion service. 
+            env_file_path (str | None): The path to the environment file. 
+            env_file_encoding (str | None): The encoding of the environment file. 
+            client (MistralAsyncClient | None): The Mistral AI client to use. 
+
+        Raises:
+            ServiceInitializationError: If an error occurs during initialization.
+        """
+        try:
+            mistralai_settings = MistralAISettings.create(
+                api_key=api_key,
+                embedding_model_id=ai_model_id,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as e:
+            raise ServiceInitializationError(f"Failed to validate Mistral AI settings: {e}") from e
+
+        if not mistralai_settings.embedding_model_id:
+            raise ServiceInitializationError("The MistralAI embedding model ID is required.")
+
+        if not client:
+            client = MistralAsyncClient(
+                api_key=mistralai_settings.api_key.get_secret_value()
+            )
+
+        super().__init__(
+            service_id=service_id or mistralai_settings.embedding_model_id,
+            ai_model_id=ai_model_id or mistralai_settings.embedding_model_id,
+            client=client,
+        )
+
+    @override
+    async def generate_embeddings(
+        self,
+        texts: list[str],
+        settings: "PromptExecutionSettings | None" = None,
+        **kwargs: Any,
+    ) -> ndarray:
+        embedding_response: EmbeddingResponse = await self.generate_raw_embeddings(texts, settings, **kwargs)
+        return array([item.embedding for item in embedding_response.data])
+
+    @override
+    async def generate_raw_embeddings(
+        self,
+        texts: list[str],
+        settings: "PromptExecutionSettings | None" = None,
+        **kwargs: Any,
+    ) -> "EmbeddingResponse":
+        """Generate embeddings from the Mistral AI service."""
+        try:
+
+            response: EmbeddingResponse = await self.client.embeddings(
```

Comment:
ideally this does return only the actual embedding, but depending on what the client returns we don't make any changes to it, but we don't want all the other stuff, so return `[item.embedding for item in response.data]`